test/e2e/framework/metrics_util.go

```diff
@@ -233,60 +233,27 @@ func (l *SchedulingMetrics) PrintJSON() string {
 	return PrettyPrintJSON(l)
 }
 
-type histogram struct {
-	Count   int
-	Buckets map[float64]int
+type Histogram struct {
+	Labels  map[string]string `json:"labels"`
+	Buckets map[string]int    `json:"buckets"`
+	Count   int               `json:"count"`
 }
 
-func (h *histogram) ConvertToRangeBuckets() {
-	keys := []float64{}
-	for k := range h.Buckets {
-		keys = append(keys, k)
-	}
-	sort.Float64s(keys)
-	lessValuesSum := 0
-	for _, k := range keys {
-		h.Buckets[k] -= lessValuesSum
-		lessValuesSum += h.Buckets[k]
-	}
-}
-
-func (h *histogram) CalculatePerc(perc float64) float64 {
-	targets := map[float64]float64{
-		perc: 0.001,
-	}
-	q := quantile.NewTargeted(targets)
-	var samples quantile.Samples
-
-	for k, v := range h.Buckets {
-		if v > 0 {
-			samples = append(samples, quantile.Sample{Value: k, Width: float64(v)})
-		}
-	}
-	q.Merge(samples)
-
-	return q.Query(perc)
-}
-
-func (h *histogram) ConvertToLatencyMetric(m *LatencyMetric) {
-	h.ConvertToRangeBuckets()
-	m.Perc50 = time.Duration(h.CalculatePerc(0.5))
-	m.Perc90 = time.Duration(h.CalculatePerc(0.9))
-	m.Perc99 = time.Duration(h.CalculatePerc(0.99))
-}
+type HistogramVec []Histogram
 
-func getNewHistogram() *histogram {
-	return &histogram{
+func newHistogram(labels map[string]string) *Histogram {
+	return &Histogram{
+		Labels:  labels,
+		Buckets: make(map[string]int),
 		Count:   0,
-		Buckets: make(map[float64]int),
 	}
 }
 
 type EtcdMetrics struct {
-	BackendCommitDuration     LatencyMetric `json:"backendCommitDuration"`
-	SnapshotSaveTotalDuration LatencyMetric `json:"snapshotSaveTotalDuration"`
-	PeerRoundTripTime         LatencyMetric `json:"peerRoundTripTime"`
-	WalFsyncFuration          LatencyMetric `json:"walFsyncFuration"`
+	BackendCommitDuration     HistogramVec `json:"backendCommitDuration"`
+	SnapshotSaveTotalDuration HistogramVec `json:"snapshotSaveTotalDuration"`
+	PeerRoundTripTime         HistogramVec `json:"peerRoundTripTime"`
+	WalFsyncDuration          HistogramVec `json:"walFsyncDuration"`
```

Comment:
Sanity check - could you confirm if all these etcd metrics are indeed vectors?

Reply:
Only `peer_round_trip_time_seconds` is a vector.
However, every histogram can be handled like a vector of size 1 containing single histogram with empty labels set.

Reply:
Ok, sgtm